### DISCUSSION TITLE
Is there any way of upgrading an extension?

### DISCUSSION THREAD

**Discussion by samirkumardas** [opencart/opencart#11651] [General]
Thank you OC Team for all your hard work.

I was looking for a way of upgrading an installed extension. In previous OC versions,  it would replace/upgrade files when an extension existed. 

But in OC 4, it shows an error message `Extension already installed!` if I try to install the same extension. It limits the way of upgrading the extension using the installer.

Is there any way of upgrading the extension?

**Comment by cartbinder** [comment]
Currently, to upgrade the extension. You need to first uninstall it via the Extension list page.
Then uninstall from the Extension installer page. Then delete the extension from the same page.
And then install it again.

**Reply by samirkumardas** [reply]
Yeah, but it has some problems like it will be removing extension DB tables or extensions data. Also,  it involves a couple of steps to follow that will be difficult for an ordinary user.

**Reply by cartbinder** [reply]
You are correct. This needs improvement. 

But one thing why uninstall would still be needed is Let's say you are adding new events in the upgrade which run only during the extension install() method. So those won't be called.
